packages/fxa-settings/src/components/UnitRowTwoStepAuth/index.tsx

```diff
@@ -0,0 +1,114 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import React, { useState } from 'react';
+import { gql, useMutation } from '@apollo/client';
+import LinkExternal from 'fxa-react/components/LinkExternal';
+import { useBooleanState } from 'fxa-react/lib/hooks';
+import AlertBar from '../AlertBar';
+import Modal from '../Modal';
+import UnitRow from '../UnitRow';
+import VerifiedSessionGuard from '../VerifiedSessionGuard';
+import { useAccount } from '../../models';
+
+export const DESTROY_TOTP_TOKEN_MUTATION = gql`
+  mutation destroyTotpToken($input: DestroyTotpToken!) {
+    destroyTotpToken(input: $input) {
+      clientMutationId
+    }
+  }
+`;
+
+export const UnitRowTwoStepAuth = () => {
+  const { totp } = useAccount();
+  const { exists, verified } = totp;
+  const [modalRevealed, revealModal, hideModal] = useBooleanState();
+  const [alertBarRevealed, revealAlertBar, hideAlertBar] = useBooleanState();
+  const [errorText, setErrorText] = useState<string>();
+  const onError = (e: Error) => {
+    setErrorText(e.message);
+    hideModal();
+    revealAlertBar();
+  };
+  const [disableTwoStepAuth] = useMutation(DESTROY_TOTP_TOKEN_MUTATION, {
+    variables: { input: {} },
+    onCompleted: () => {
+      hideModal();
+      revealAlertBar();
+    },
+    onError,
+    ignoreResults: true,
+    update: (cache) => {
+      cache.modify({
+        fields: {
+          account: (existing) => {
+            return { ...existing, totp: false };
+          },
+        },
+      });
+    },
+  });
+
+  return (
+    <UnitRow
+      header="Two-step authentication"
+      headerValueClassName={exists ? 'text-green-800' : ''}
+      headerValue={exists ? 'Enabled' : 'Not Set'}
+      route="/beta/settings/two_step_authentication"
+      ctaText={exists ? 'Change' : 'Add'}
+      secondaryCtaText={exists ? 'Disable' : undefined}
+      secondaryButtonClassName={
+        exists ? 'bg-red-500 text-white border-red-600' : ''
+      }
+      revealSecondaryModal={exists ? revealModal : undefined}
+    >
+      <p className="text-sm mt-3">
+        Prevent someone else from logging in by requiring a unique code only you
+        have access to.
+      </p>
+      {modalRevealed && (
+        <VerifiedSessionGuard onDismiss={hideModal} onError={onError}>
+          <Modal
+            onDismiss={hideModal}
+            onConfirm={disableTwoStepAuth}
+            headerId="some-id"
+            descId="some-desc"
+          >
+            <h2 id="some-id" className="font-bold text-xl text-center mb-2">
+              Disable Two-stop Authentication?
```

Comment:
The spelling/capitalization needs to be updated, but also this modal should more closely resemble the Figma doc. Looks like we need to use the Red disable button that says "Disable", instead of a blue button that says "Confirm", and the "replacing your recovery codes" link looks off. The button horizontal spacing is also off, but I am noticing this on main too, so it could be addressed later if not here.

Figma:

<img width="444" alt="Screen Shot 2020-08-26 at 2 27 13 PM" src="https://user-images.githubusercontent.com/6392049/91341909-449ec800-e7a8-11ea-8af9-eef56298f7b9.png">

Actual:

<img width="538" alt="Screen Shot 2020-08-26 at 2 26 43 PM" src="https://user-images.githubusercontent.com/6392049/91341873-32bd2500-e7a8-11ea-841b-9e8d8646725d.png">

Reply:
I updated the text of and color of the disable button, I think we should file a new bug for fixing the default styles of the modal since it affects other parts of the app, and seems out of scope for this issue. Filed https://jira.mozilla.com/browse/FXA-2486